crates/driver/src/boundary/notifier/mod.rs

```diff
@@ -0,0 +1,58 @@
+mod liquorice;
+
+use {
+    crate::{
+        boundary::notifier::liquorice::LiquoriceNotifier,
+        domain::competition::solution::settlement::Settlement,
+        infra,
+    },
+    anyhow::Result,
+    futures::future::join_all,
+};
+
+/// Trait for notifying liquidity sources about auctions and settlements
+#[async_trait::async_trait]
+pub trait LiquiditySourcesNotifying: Send + Sync {
+    async fn notify_before_settlement(&self, settlement: &Settlement) -> Result<()>;
+}
+
+pub struct Notifier {
+    inner: Vec<Box<dyn LiquiditySourcesNotifying>>,
+}
+
+impl Notifier {
+    pub fn try_new(
+        config: &infra::notify::liquidity_sources::Config,
+        chain_id: u64,
+    ) -> Result<Self> {
+        let mut inner: Vec<Box<dyn LiquiditySourcesNotifying>> = vec![];
+
+        if let Some(liquorice) = &config.liquorice {
+            inner.push(Box::new(LiquoriceNotifier::new(liquorice, chain_id)?));
+        }
+
+        Ok(Self { inner })
+    }
+}
+
+#[async_trait::async_trait]
+impl LiquiditySourcesNotifying for Notifier {
+    async fn notify_before_settlement(&self, settlement: &Settlement) -> Result<()> {
```

Comment:
A bit odd to have the `_before_settlement` suffix here as the trait can't enforce when it gets called.

Reply:
I named it this way to be able to add more notification logic in the future - after settlement, before auction and etc.

Would `notify_on_settlement` be okay?

Reply:
Changed to `settlement`